tests/conftest.py

```diff
@@ -45,3 +45,26 @@ def copy_case(path, config_file):
 
     with tmpdir.as_cwd():
         yield copy_case
+
+
+@pytest.fixture()
+def unused_tcp_port(unused_tcp_port, monkeypatch):
```

Comment:
Are we sure we want to overwrite this fixture (coming from pytest-asyncio, right?) for all of ERT?

Shouldn't localhost be a value that can be set _somewhere_ and ultimately reach `custom_host` of `find_available_port`?

Reply:
The concern here was that although it is easy for the low level tests to add `hostname`, the same may not be true for e.g. integration tests. Expanding the api exclusively to cater for running the tests locally seemed unnecessary. I'll admit I did not go through all the tests that uses `unused_tcp_port`, but I can.

Are there any obvious scenarios where we would _want_ the public ip address out of `_get_ip_address` in a test? (I do understand that even with a "no" to that question, it's still debatable to wrap the fixture for the entire ERT test setup..)

> (coming from pytest-asyncio, right?)

Correct

Reply:
> Are there any obvious scenarios where we would _want_ the public ip address out of `_get_ip_address` in a test?

From a different angle: Are there scenarios where a test would use `unused_tcp_port` and fail if the system binds it to localhost, i.e. is it likely to hurt us to wrap this fixture?

The obvious alternative is to introduce our own fixture, e.g. `ert_unused_tcp_port`, but then 1) lots of existing tests must be patched, and 2) we risk having new tests use the original `unused_tcp_port` and miss this fix.

Reply:
> 2) we risk having new tests use the original unused_tcp_port and miss this fix.

This is valid point. It wasn't obvious what the issue was based on the error when a test failed. It could be that test documentation on how we should run tests that opens sockets must be setup would improve this (perhaps make it searchable by adding typical error message)

Reply:
> Are there any obvious scenarios where we would want the public ip address out of _get_ip_address in a test?

Some test will test our secure websocket/http communication, and I'm not entirely sure if you can have e.g. websockets accept certs made for localhost? If you do, then no—we probably want localhost everywhere.

> Just for reference, the unused_tcp_port is used in 16 tests from what I could see

Did you follow fixtures using this fixture? The search has to be recursive.

> I would say it's reasonable that an EvaluatorServerConfig should accept a host.

I agree.

Reply:
> Did you follow fixtures using this fixture? The search has to be recursive.

My search was not exhaustive it seems, there were about 30 more tests that used `unused_tcp_port`. A single fixture used it, but in combination with `localhost`

Reply:
> > Are there any obvious scenarios where we would want the public ip address out of _get_ip_address in a test?
> 
> Some test will test our secure websocket/http communication, and I'm not entirely sure if you can have e.g. websockets accept certs made for localhost? If you do, then no—we probably want localhost everywhere.

Mmmm...  could you point to any of these tests so we could check?

It should be the original (unchanged) "fixture-object" from `pytest-asyncio` which is given to the test-method. Our wrapper implements a side-effect - perhaps we can just parametrize our wrapper to avoid this side-effect if it is not desired?

Reply:
> Mmmm... could you point to any of these tests so we could check?

I think it's more or less the default. But a good compromise would be one integration test with secure comms, and all others without.